examples/flutter_gallery/lib/demo/material/leave_behind_demo.dart

```diff
@@ -214,6 +227,31 @@ class _LeaveBehindListItem extends StatelessWidget {
           else
             _handleDelete();
         },
+        confirmDismiss: (DismissDirection dismissDirection) async {
```

Comment:
This could be simplified just a little bit more:
```
confirmDismiss: !confirmDismiss ? null : (DismissDirection dismissDirection) async {
  switch(dismissDirection) {
    case DismissDirection.endToStart:
      if (await _showConfirmationDialog(context, 'archive'))
        _handleArchive();
      break;
    case DismissDirection.startToEnd:
      if (await _showConfirmationDialog(context, 'delete'))
        _handleDelete();
      break;
    case DismissDirection.horizontal:
    case DismissDirection.vertical:
    case DismissDirection.up:
    case DismissDirection.down:
      assert(false);
  }
},
```